### PR TITLE
User algebraics

### PR DESCRIPTION
**Pull request by apozharski**

### CODE REVIEW

File: examples/matlab/gene_regulation/irma_model.m [hunk 1]
```diff
@@ -52,10 +52,29 @@
...
+        else
+            beta = SX.sym('beta', 2);
+            model.f_alg = beta - [alpha(1)*(1-alpha(7)); alpha(2)*(1-alpha(5))];% lifted equations
+            s = [1, alpha(6);
+             1, beta(1);
+             1, alpha(3);
+             alpha(2), beta(2);
+             1, alpha(4)];
+        end
+        model.z_user = beta;
```
Comment (nurkanovic) [inline examples/matlab/gene_regulation/irma_model.m:72]: I was thinking to name the user algebraic variables just  `z`, and internally when they get collected to call them
`z_all` 

All other variables defined by the user that enter `model` do not have subscripts (except `v_global`) and
for `f_alg`, i would prefer `g_z` and all them together `g_z_all` (as it already is)

File: src/matlab/FiniteElement.m [hunk 2]
```diff
@@ -282,6 +283,17 @@
...
                 end
+
+                % add user variables
+                z_user = define_casadi_symbolic(settings.casadi_symbolic_mode,...
+                                                ['z_user_' num2str(ctrl_idx-1) '_' num2str(fe_idx-1) '_' num2str(ii)],...
+                                                dims.n_z_user);
+                obj.addVariable(z_user,...
+                                'z_user',...
+                                -inf*ones(dims.n_z_user,1),...
+                                inf*ones(dims.n_z_user,1),...
```
Comment (nurkanovic) [inline src/matlab/FiniteElement.m:320]: I would make it possible for the user to add lbz and ubz box constraints on the algebraic variables

File: examples/matlab/gene_regulation/irma_model.m (hunk 1 above)
Comment (apozharski) [reply examples/matlab/gene_regulation/irma_model.m]: Rename is done.

**Review by nurkanovic — APPROVED** [review]
Great work @apozharski 🚀

**Pull request merged**